PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/AzureRunProfileState.java

```diff
@@ -59,20 +60,40 @@ public ExecutionResult execute(Executor executor, @NotNull ProgramRunner program
         Map<String, String> telemetryMap = new HashMap<>();
         Observable.fromCallable(
             () -> {
+                sendOperStart();
+                sendOperInfo(telemetryMap);
                 return this.executeSteps(processHandler, telemetryMap);
             }).subscribeOn(SchedulerProviderFactory.getInstance().getSchedulerProvider().io()).subscribe(
             (res) -> {
+                sendOperEnd();
                 this.sendTelemetry(telemetryMap, true, null);
                 this.onSuccess(res, processHandler);
             },
             (err) -> {
                 err.printStackTrace();
+                sendOperError(ErrorType.systemError, err.getMessage(), telemetryMap);
+                sendOperEnd();
                 this.onFail(err.getMessage(), processHandler);
                 this.sendTelemetry(telemetryMap, false, err.getMessage());
             });
         return new DefaultExecutionResult(consoleView, processHandler);
     }
 
+    protected void sendOperStart(){
+    }
```

Comment:
make it abstract?

Reply:
If make it abstract, several classes will have compile error.
This commit just for webapp.
After fix all productions, then I will make it a abstract method